plugins/filter_aws/aws.c

```diff
@@ -370,6 +400,208 @@ static int get_vpc_metadata(struct flb_filter_aws *ctx)
     return ret;
 }
 
+void flb_filter_aws_tags_destroy(struct flb_filter_aws *ctx)
+{
+    size_t i;
+    if (!ctx) {
+        return;
+    }
+    if (ctx->tag_keys) {
+        for (i = 0; i < ctx->tags_count; i++) {
+            if (ctx->tag_keys[i]) {
+                flb_sds_destroy(ctx->tag_keys[i]);
+            }
+        }
+        flb_free(ctx->tag_keys);
+        ctx->tag_keys = NULL;
+    }
+    if (ctx->tag_values) {
+        for (i = 0; i < ctx->tags_count; i++) {
+            if (ctx->tag_values[i]) {
+                flb_sds_destroy(ctx->tag_values[i]);
+            }
+        }
+        flb_free(ctx->tag_values);
+        ctx->tag_values = NULL;
+    }
+    if (ctx->tag_keys_len) {
+        flb_free(ctx->tag_keys_len);
```

Comment:
size_t can't be freed. The memory needed for it will be accounted for when the context struct is allocated.

Reply:
`tag_keys_len` is an array of `size_t` values. It is `size_t *tag_keys_len`, so I don't think memory is going to be created for the whole array at the allocation for the `ctx`. Am I misunderstanding something here?

Reply:
apologies, I misread this code. You're right.